SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriverRPC.cpp

```diff
@@ -164,7 +164,7 @@ void FSpatialNetDriverRPC::ProcessReceivedRPCs()
 	// MulticastReceiver->
 }
 
-TArray<FWorkerComponentData> FSpatialNetDriverRPC::GetRPCComponentsOnEntityCreation(const Worker_EntityId EntityId)
+TArray<Worker_ComponentId> FSpatialNetDriverRPC::GetRPCComponentIds()
```

Comment:
Where is this new method actually used? Was it the idea that you could create these components in advance, with empty state, and just update them when fleshing out the skeleton? Could/should we actually do that?

Reply:
This is used during skeleton entity construction; this comment does show that it'd be better to move the usages to the EntityFactory so they are closer to the other usages.
Unrelated, but do we need to update them when fleshing out? Won't the RPC system recognize that there is an RPC endpoint and write straight to it without queueing?

Reply:
It seems like it's used to set another static array. Although if we just have a const static array can we just make it that in file name scope? Or something that has less indirection? At least one of the function level static arrays isn't needed.

Reply:
It is utilized in https://github.com/spatialos/UnrealGDK/pull/3160/files#diff-e27dcbf8c7ac7ba4e0fdcc27bc0ac20ad9e6157ee032cd6ba2ce9ad1f77abd28R408-R409 during skeleton entity creation.

Reply:
I'd still raise the same thing though. Instead of having a static function return a const static array and store that in another static array (or just read it in the case of the thing you linked), can we just have a const static array as a member?